scripts/BUILD.bazel

```diff
@@ -0,0 +1,46 @@
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary", "kt_jvm_library", "kt_jvm_test")
+load("//scripts:script_assets.bzl", "generate_assets_list_from_text_protos")
+
+REGEX_PATTERN_CHECK_ASSETS = generate_assets_list_from_text_protos(
+    name = "script_assets",
+    file_content_validation_file_names = [
+        "file_content_validation_checks",
+    ],
+    filepath_pattern_validation_file_names = [
+        "filename_pattern_validation_checks",
+    ],
+)
+
+kt_jvm_binary(
+    name = "pattern_validation_check",
+    testonly = True,
+    data = REGEX_PATTERN_CHECK_ASSETS,
+    main_class = "org.oppia.android.scripts.regex.RegexPatternValidationCheckKt",
+    runtime_deps = ["//scripts/src/java/org/oppia/android/scripts/regex:pattern_validation_lib"],
+)
+
+kt_jvm_test(
+    name = "RegexPatternValidationCheckTest",
+    testonly = True,
+    srcs = ["src/test/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt"],
+    data = REGEX_PATTERN_CHECK_ASSETS,
+    deps = [
+        "//scripts/src/java/org/oppia/android/scripts/regex:pattern_validation_lib",
+        "//testing:assertion_helpers",
+        "//third_party:com_google_truth_truth",
+        "//third_party:org_jetbrains_kotlin_kotlin-test-junit",
+    ],
+)
```

Comment:
@BenHenning My only intention to add the test rule here was that because, it needs the `REGEX_PATTERN_CHECK_ASSETS` as a `data` dependency. Hence, it has to be at the root level BUILD. Also, to maintain uniformness, it seems good to add all the test rules here only. Let me know your thoughts.

Reply:
Tests should always be next to their production implementations (we don't do this today elsewhere because we're in a mid-migration state from Gradle to Bazel, but the design proposed here is the recommended best practice). Could we move ``REGEX_PATTERN_CHECK_ASSETS`` to a Bazel file within scripts/regex & import it?